include/perf_watcher.h

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#pragma once
+
+#include <linux/perf_event.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+struct PerfWatcherOptions {
+  bool is_kernel;
+  bool is_freq;
+};
+
+typedef struct PerfWatcher {
+  const char *desc;
+  uint64_t sample_type;
+  int type;
+  unsigned long config;
+  union {
+    uint64_t sample_period;
+    uint64_t sample_frequency;
+  };
+  int profile_id;
+  // perf_event_open configs
+  struct PerfWatcherOptions options;
+  int count_id;
```

Comment:
`sample_count_type_id` ?

Reply:
Changed to `count_sample_type_id` since it is only used in one place, but that place needs to be informative :)